### ISSUE TITLE
Add a way to include all minor MC versions

### ISSUE THREAD

**Issue by Andre601** [Kir-Antipov/mc-publish#25]
From what I understand do you currently have to define each MC version separately as a `game-versions` option when the automatic resolution isn't a usable option.

While this may not be that big of a deal for a mod, as they usually are only supporting a limited range of versions per release (i.e. 1.19, 1.19.1 and 1.19.2), is this not the same for Plugins and resource packs, which Curseforge supports and Modrinth does too now.

For a plugin is it not unusual to support any versions from 1.8 up to 1.19.2 as the Plugin API is a lot more consistent across versions compared to the more fluid modding API and as long as the plugin isn't using features introduced in a specific version is the plugin compatible with pretty much every MC version available.
Resource packs also suffer from this issue.

This can result in a `game-versions` list similar to this nightmare:
```yaml
game-versions: |
  1.8.8
  1.9
  1.9.1
  1.9.3
  1.10
  1.11
  1.11.1
  1.12
  1.12.1
  1.12.2
  1.13
  1.13.1
  1.13.2
  1.14
  1.14.1
  1.14.2
  1.14.3
  1.14.4
  1.15
  1.15.1
  1.15.2
  1.16
  1.16.2
  1.16.3
  1.16.4
  1.16.5
  1.17
  1.17.1
  1.18
  1.18.1
  1.18.2
  1.19
  1.19.1
  1.19.2
```
Because of this would I like to suggest a feature, that would allow defining a major version which would include all minor ones.
The idea is to allow a pattern like `1.19.x` and the action would automatically include all the known minor versions available.

This would turn the above list into this:
```yaml
game-versions: |
  1.8.8
  1.9.x
  1.10
  1.11.x
  1.12.x
  1.13.x
  1.14.x
  1.15.x
  1.16.x
  1.17.x
  1.18.x
  1.19.x
```
As you can see would the list already be a quite shorter compared to its previous version.

**Comment by Kir-Antipov** [comment]
I don't really want to make the project more complex than it is already, so a custom version format that users need to additionally learn does not seem like a way to go for me. Also, it's a temporary solution - 10 more years in the future and you'll make another issue complaining about the need to specify every minor from 8 to 50 :)

What I think should be done:

1) I can make a new `game-version` input, where you can specify a Minecarft version that will be expanded into a version range via existing mechanics of the `version-resolver` input
2) Refactor the `version-resolver` input, because at the moment it only works with builds

So, your configuration can be turned into something like this:

```yaml
game-version: 1.8.8
version-resolver: major | minor | build # Or something like that, I think I will turn this input into enum so it can be fine-tuned with absolute ease
```

**Comment by Andre601** [comment]
Well, another option could be a syntax like `game-versions: 1.8.8 - 1.19.2` to indicate "Include all versions between 1.8.8 and 1.19.2"
And have a separate setting to set if dev-builds should be included or not (Like using the version-resolver as you mentioned).

Either way, some option to keep the amount of versions to list at a minimum would be desirable...

**Comment by Kir-Antipov** [comment]
Thought about this issue for quite some time. I suppose, it is time for `version-resolver` to go. It was a bad API I made because I did not want to deal with things like `3D Shareware v1.34` *(yes, this is a valid Minecraft version. I understand, that this was an April Fools' joke, but now it's just a huge pain in the back to deal with)*

Modloaders that mc-publish supports out of the box allow users to specify not a single version, but a version range, therefore I already can extract this information from their metadata files. The problem was and is that I cannot use existing solutions to parse this ranges because of the Minecraft versioning weirdness, but yeah, I guess it just should be done

If I finally get it to work, it will look like this:
```yaml
# Fabric-like version ranges, which are close to https://devhints.io/semver
game-versions: >=1.8.8 <=1.19.2
game-versions: 1.8.8 - 1.19.2

# Forge-like version ranges, because I need to support them anyway
game-versions: [1.8.8, 1.19.2]

# Formats can be mixed
game-versions: >= 1.8.8 <1.16 || [1.16, 1.19.2]

# Versions can still be listed separately on different lines,
# because all of them will be treated as ranges concatenated via OR operation
game-versions: |
  1.8.8
  1.9.x
  1.10
  1.11.x
  1.12.x
  1.13.x
  1.14.x
  1.15.x
  1.16.x
  1.17.x
  1.18.x
  1.19.x
```

But now I need another way for users to specify if they want for snapshots to appear on their project page or not. I'm open for suggestions :D

**Comment by Andre601** [comment]
Maybe a simple `include-snapshots` option?
Alternatively just have it included if explicitly defined, because plugins never (or only very rarely) support dev builds and mods usually do define snapshots anyways, right?

**Comment by Kir-Antipov** [comment]
> Maybe a simple `include-snapshots` option?

Yeah, should do it, I think, because even if a mod, or a plugin explicitly specifies snapshots as something that they should support, it's not always the case that their developers want a bunch of pre-release Minecraft versions to clutter their project page.

**Comment by Kir-Antipov** [comment]
Implemented in [v3.3](https://github.com/Kir-Antipov/mc-publish/releases/tag/v3.3).
